addons/pos_cache/models/pos_cache.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from datetime import datetime
+import cPickle
+
+from openerp import models, fields, api
+
+class pos_cache(models.Model):
+    _name = 'pos.cache'
+
+    cache = fields.Binary()
+    product_domain = fields.Binary()
+    product_fields = fields.Binary()
+
+    config_ids = fields.One2many('pos.config', 'cache_id', 'Config')
+    compute_user_id = fields.Many2one('res.users', 'Cache compute user',
+                                      help="User used to compute the cache content and thus avoid the bad practice of using admin in multi-company")
+
+    @api.model
+    def create(self, values):
+        import pudb; pu.db
+        values.update({
+            'product_domain': cPickle.dumps(values['product_domain'], protocol=cPickle.HIGHEST_PROTOCOL),
+            'product_fields': cPickle.dumps(values['product_fields'], protocol=cPickle.HIGHEST_PROTOCOL),
+        })
+
+        return super(pos_cache, self).create(values)
+
+    @api.model
+    def refresh_all_caches(self):
+        self.env['pos.config'].search([('state', '=', 'active')]).recompute_cache()
+
+    @api.one
+    def refresh_cache(self):
+        products = self.env['product.product'].search(cPickle.loads(self.product_domain))
+        prod_ctx = products.with_context(pricelist=self.config_ids[0].pricelist_id.id, display_default_code=False)
+        if self.compute_user_id:
+            prod_ctx = prod_ctx.sudo(self.compute_user_id.id)
+        res = prod_ctx.read(cPickle.loads(self.product_fields))
+        datas = {
+            'cache': cPickle.dumps(res, protocol=cPickle.HIGHEST_PROTOCOL),
+        }
+
+        self.write(datas)
+
+class pos_config(models.Model):
+    _inherit = 'pos.config'
+
+    #Use a related model to avoid the load of the cache when the pos load his config
+    cache_id = fields.Many2one('pos.cache', 'Cache')
+
+    cache_time = fields.Datetime(related='cache_id.write_date', string='Last cache compute')
+    cache_compute_user_id = fields.Many2one('res.users', related='cache_id.compute_user_id')
+
+    @api.multi
+    def unlink(self):
+        self.cache_id.unlink()
+        return super(pos_config, self).unlink()
```

Comment:
ondelete is used to determine the behavior of what happens with the current record when a linked many2one record gets deleted. So in this case we can't use it, because we want to delete the pos_cache when pos_config gets deleted, not delete the pos_config when the pos_cache gets deleted.